internal/helm/helm_test.go

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Authors of Cilium
+
+package helm
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/blang/semver/v4"
+)
+
+func TestResolveHelmChartVersion(t *testing.T) {
+	type args struct {
+		versionFlag        string
+		chartDirectoryFlag string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    semver.Version
+		wantErr bool
+	}{
+		{
+			name:    "valid-version",
+			args:    args{versionFlag: "v1.11.5", chartDirectoryFlag: ""},
+			want:    semver.Version{Major: 1, Minor: 11, Patch: 5},
+			wantErr: false,
+		},
+		{
+			name:    "missing-version",
+			args:    args{versionFlag: "v0.0.0", chartDirectoryFlag: ""},
+			wantErr: true,
+		},
+		{
+			name:    "invalid-version",
+			args:    args{versionFlag: "random-version", chartDirectoryFlag: ""},
+			wantErr: true,
+		},
+		{
+			name:    "valid-chart-directory",
+			args:    args{versionFlag: "", chartDirectoryFlag: "./test"},
```

Comment:
Nit: I'd probably name the directory `testdata` which is a bit less generic and seems to be a common name used in Go projects for files consumed by tests:

```suggestion
			args:    args{versionFlag: "", chartDirectoryFlag: "./testdata"},
```